File: ios/ConnectionHandler.m
//
//  ConnectionHandler.h
//  TerraRtReact
//
//  Created by Elliott Yu on 16/01/2026.
//  Copyright © 2026 Facebook. All rights reserved.
//


#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import <Foundation/Foundation.h>

@interface RCT_EXTERN_MODULE(ConnectionHandler, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)
RCT_EXTERN_METHOD(update:(BOOL)update)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/ConnectionHandler.swift
//
//  ConnectionHandler.swift
//  TerraRtReact
//
//  Created by Elliott Yu on 19/05/2023.
//  Copyright © 2023 Facebook. All rights reserved.
//

import Foundation

@objc(ConnectionHandler)
open class ConnectionHandler: RCTEventEmitter{
    @objc public static var emitter: RCTEventEmitter!
    
    override init(){
        super.init()
        ConnectionHandler.emitter = self
    }

  @objc open func update(_ connection: Bool) {
        sendEvent(withName: "ConnectionUpdate", body: connection)
    }
    
    open override func supportedEvents() -> [String]! {
      return ["ConnectionUpdate"]
    }
}
